scripts/azureml-assets/azureml/assets/validate_assets.py

```diff
@@ -145,6 +148,25 @@ def validate_environment_name(asset_config: assets.AssetConfig) -> int:
     return error_count
 
 
+def validate_dockerfile(environment_config: assets.EnvironmentConfig) -> int:
+    """Validate Dockerfile.
+
+    Args:
+        environment_config (EnvironmentConfig): Environment config.
+
+    Returns:
+        int: Number of errors.
+    """
+    error_count = 0
+    dockerfile = environment_config.get_dockerfile_contents()
+
+    if DOCKERFILE_IMAGE_PATTERN.match(dockerfile):

```

Comment:
`match()` looks only at the beginning of a string. I think `search()` would be a better fit here:
```suggestion
    if DOCKERFILE_IMAGE_PATTERN.search(dockerfile):
```